lib/iris/fileformats/pp.py

```diff
@@ -2404,6 +2416,11 @@ def save_pairs_from_cube(cube, field_coords=None, target=None):
 
     # Save each named or latlon slice2D in the cube
     for slice2D in cube.slices(field_coords):
+        # Attach an extra cube "fill_value" property, allowing the save rules
+        # to deduce MDI more easily without realising the data.
+        # NOTE: it is done this way because this property may exist in future.
+        slice2D.fill_value = _data_fill_value(slice2D)
```

Comment:
Perhaps this could be avoided by making the code in `_data_fill_value()` into a set of save rules to determine `pp.bmdi`.